### ISSUE TITLE
C++20 modules

### ISSUE THREAD

**Issue by 1vanK** [doxygen/doxygen#9116]
Apologies if this question has already been asked, but I couldn't find it. C++20 modules are already well supported in VS (with the exception of IntelliSense), so I plan to use C++20 modules in new projects. For me support of Doxgyen is important  (xmls can be used to generate automatic bindings to script langs). Is there any work being done on module support at the moment. And if so, at what stage?

**Comment by albert-github** [comment]
There are a number of C++20 features that have already been included in doxygen.
There are a number of issues Regarding C++20 that are still open (see e.g. #6702, #7261, #7799, #8772)

Feel free to create proposed patches for required C++20 features.

**Comment by 1vanK** [comment]
I already participate in several open source projects, so I will not be able to participate in one more, unfortunately.

**Comment by 1vanK** [comment]
Also link to standart https://en.cppreference.com/w/cpp/language/modules

**Comment by 1vanK** [comment]
Essentially, the compiler converts the module into h and cpp files. The header file will contain what is marked as `export` in moduile.

**Comment by Lveyou** [comment]
It would be great if someone could solve this problem. I usually use it like this:

B.ixx
```
module;
//old style include
#include <old.h>
export module B;//export module, name is B

import A;//A is module
export import C;//C is module, and transpond C

//export namespace all
export namespace ns{
	void func_a()
	{
	}

	void func_b();

	class B
	{
		//...
	};

}

//not export namespace
namespace ns{
	//but there export
	export void func_c()
	{
	}
	
	export class C
	{
		//...
	};
}

//private part, like B.cpp
module: private;

namespace ns{

	void func_b()
	{
	}

}
```

C.cpp
```
module;
#include <zzz.h>
module B;//module B impl，like B.cpp

import D;

namespace ns{
	
	void B::B():_x{0}
	{}
	
	void B::func_x()
	{
		print(_x);
	}
	
}


```

**Comment by albert-github** [comment]
@Lveyou 

In your description some information is missing:
- version of doxygen you are using
- the problem you face, I think it is regarding the support of modules.

In general:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.
- Please also specify the **full** doxygen version used (`doxygen -v`).

**Comment by Lveyou** [comment]
I'm using version 1.9.5, and if the latest version correctly supports the C++ module, that's enough.
A simple example is as follows:

[doxygen_test_module_cpp.zip](https://github.com/doxygen/doxygen/files/9886018/doxygen_test_module_cpp.zip)

**Comment by doxygen** [comment]
With this [commit](https://github.com/doxygen/doxygen/commit/6d80fc7e5d03c259b1a7280972e0b28884217655) support for C++20 modules is introduced. Please check if it is working as expected for you. Do not close this report, this will be done automatically with the next official release.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
